### ISSUE TITLE
guestregister and ca-certificates.service race problem

### ISSUE THREAD

**Issue by ld9379435** [SUSE-Enceladus/cloud-regionsrv-client#151]
in some suse distro，guestregister and ca-certificates.service has a race problem like that：https://www.suse.com/support/kb/doc/?id=000021252 ;

I noticed in the new version, there is a commit at https://github.com/SUSE-Enceladus/cloud-regionsrv-client/commit/3007f85e2f591db4e4cf1399fb97f2a396102c51 that aims to alleviate this race problem.

However, I have a question regarding the import_smtcert_12 function in the file at https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/3962bc4bb356eb9adaf8b312ced7a167b029ca5b/lib/cloudregister/registerutils.py#L1012. When smt.write_cert(key_chain) is executed, it triggers the monitoring of ca-certificates.path, which subsequently invokes the ca-certificates.service to run /usr/sbin/update-ca-certificates for updating the certificates. But then, in the update_ca_chain function, it seems that this update operation is called again, thus causing this issue. I am puzzled as to why this update operation needs to be invoked twice.

**Comment by rjschwei** [comment]
It doesn't need to be invoked twice. The certs used to be written to a different location that was not monitored because there was a bug somewhere else. When the location where we write the certs changed the now superfluous invocation to refresh the certs was not removed. At the same time, if we remove the invocation than we depend on the monitoring to work properly and if for some reason it does not we create problems. As such it is not as simple as removing the invocation. We'd have to verify that the new certs got picked up by the system and if not run the command.

**Comment by schaefi** [comment]
@rjschwei 

> The certs used to be written to a different location that was not monitored because there was a bug somewhere else. 

I guess this part of the problem is gone with the removal of the import_smtcert_12() method which also seems to be the target for the handling of a cert location that was not monitored. Would you agree ?

Given the above is correct we can therefore assume that the only left location for certs is also expected to be monitored correctly.
If you would also agree to this we can make a first step and drop the explicit invocation of `update_ca_chain(['update-ca-certificates'])` from `import_smt_cert()`

To my understanding this should fix the race condition ?

Next you mentioned 

> if we remove the invocation than we depend on the monitoring to work properly and if for some reason it does not we create problems. 

This implies you don't trust the fancy monitoring implementation :) Fair enough. So once step 1 is done we can add another PR that adds a method to verify if the cert was correctly included and if not call `update-ca-certificates` manually as a last resort action. If this sounds acceptable I would have some questions:
* If `update-ca-certificates` is not allowed to be called in parallel, what is the suggested best way to check if it's not running when we see the need to call it ?
* What is the best way to identify the cert after import. I was looking at `find /etc/pki/trust/anchors -name "*.pem" | xargs -I{} openssl x509 -issuer -enddate -noout -in {}` ?

Thanks
